### PR TITLE
Clean up absl-base and time external binds.

### PR DESCRIPTION
**Pull request by veblush**
Landing https://github.com/grpc/grpc/pull/29093 with rebasing it onto master.

### CODE REVIEW

File: bazel/grpc_deps.bzl [hunk 1]
```diff
@@ -55,21 +55,6 @@ def grpc_deps():
         actual = "@upb//:generated_code_support__only_for_generated_code_do_not_use__i_give_permission_to_break_me",
     )
 
-    native.bind(
-        name = "absl",
```
Comment (drfloob) [inline bazel/grpc_deps.bzl:59]: Please double check that this doesn't require a cherrypick.

**Comment by drfloob** [comment]
Similarly, please ensure no conflicts in `fix_build_deps.py`

File: bazel/grpc_deps.bzl (hunk 1 above)
Comment (veblush) [reply bazel/grpc_deps.bzl:59]: Yep I just tested!

**Comment by veblush** [comment]
> Similarly, please ensure no conflicts in `fix_build_deps.py`

I believe this is already verified by the sanity test.

**Pull request merged**